modules/lex_calendar/templates/calendar-page.html.twig

```diff
@@ -97,28 +146,66 @@ window.addEventListener( 'load', function(){
+
+              $('.fc-list-heading[data-date="'+ now.format('YYYY-MM-DD') +'"]').css('display', 'table-row');
+              $('.fc-list-heading[data-date="'+ now.format('YYYY-MM-DD') +'"]').append($listBlock);
+              now.add(1, 'days');
+            }
+          }else {
+            for (var i = 0; i < events.length; i++) {
+              var event = events[i];
+              var name = event.title;
+              var now = event.start.clone();
 
-          // append to calendar day
-          $('.fc-list-heading[data-date="'+ now.format('YYYY-MM-DD') +'"]').append($listBlock);
-          now.add(1, 'days');
+              if (event.start.format('h:mm a')=='12:00 p.m.') {
+                startTime = 'noon';
+              }else if(event.start.format('mm')=='00') {
+                startTime = event.start.format('h a')
+              }else {
+                startTime = event.start.format('h:mm a');
+              }
+
+
+              if (event.relatedDepartments == $('#calendar-filter').val()) {
+                var $listBlock = $(
+                  '<div class="list-event-container" id="list-' + event.id + '"><a href="' + event.url + '"><span class="fc-day-grid-event fc-event fc-start fc-not-end dot-event list-dot " style="background-color:' + event.color +'" title=""></span><div class="list-text-container"><span class="list-event-title">' + event.title + '</span><span class="list-event-date">' + event.start.format('dddd, MMM. D, YYYY') + '</span><span class="list-event-time">' + startTime + '</span><span class="details-link">Details <i class="fas fa-chevron-right"></i></div></a></div>');
+                $('.fc-list-heading[data-date="'+ now.format('YYYY-MM-DD') +'"]').css('display', 'table-row');
+                 $('.fc-list-heading[data-date="'+ now.format('YYYY-MM-DD') +'"]').addClass('displayList');
+                $('.fc-list-heading[data-date="'+ now.format('YYYY-MM-DD') +'"]').append($listBlock);
+              }else {
+                if ($('.fc-list-heading[data-date="'+ now.format('YYYY-MM-DD') +'"]').css('display', 'table-row')) {
+                  $('.fc-list-heading[data-date="'+ now.format('YYYY-MM-DD') +'"]').css('display', 'none');
+                }
+              }
+            }
+          }
         }
+        $('#calendar-filter').change(function() {
+          $('#sidebar-calendar').find('.fc-list-heading').removeClass('displayList');
+          $('#sidebar-calendar').find('.list-event-container').remove();
+          listBuild();
+        });
       }
     });
   })(jQuery);
```

Comment:
I would try to clean this file up. Lots of repeated code for the same or similar results. I'd feel better merging this if we can clean it up.